services/src/main/java/io/scalecube/services/inject/ScaleCubeServicesProvider.java

```diff
@@ -0,0 +1,101 @@
+package io.scalecube.services.inject;
+
+import io.scalecube.services.Microservices;
+import io.scalecube.services.ServiceDefinition;
+import io.scalecube.services.ServiceInfo;
+import io.scalecube.services.ServiceProvider;
+import io.scalecube.services.ServicesProvider;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * Default {@link ServicesProvider}.
+ */
+public class ScaleCubeServicesProvider implements ServicesProvider {
+
+  private final Function<Microservices, Collection<ServiceInfo>> serviceFactory;
+
+  // lazy init
+  private Collection<ServiceInfo> services;
```

Comment:
Can we avoid lazy init? Probably, need to apply thread-safe stuff for it.

Reply:
See suggestions below.

Reply:
Thank you for the remark. The original implementation was thread safe, but 
- this service was not intended to be made public api
- its use is limited to the inside of ScaleCube

So I decided not to overcomplicate the implementation